registry-library/library/library.go

```diff
@@ -55,16 +55,22 @@ var (
 	DevfileAllMediaTypesList = []string{DevfileMediaType, DevfilePNGLogoMediaType, DevfileSVGLogoMediaType, DevfileVSXMediaType, DevfileArchiveMediaType}
 )
 
-// GetRegistryStacks returns the list of stacks, more specifically
-// it gets the content of the index (index.json) of the specified registry
-// for listing the stacks
-func GetRegistryStacks(registry string) ([]indexSchema.Schema, error) {
+// GetRegistry returns the list of stacks and/or samples, more specifically
+// it gets the stacks and/or samples content of the index of the specified registry
+// for listing the stacks and/or samples
+func GetRegistry(registry string, devfileType indexSchema.DevfileType) ([]indexSchema.Schema, error) {
 	// Call index server REST API to get the index
 	urlObj, err := url.Parse(registry)
 	if err != nil {
 		return nil, err
 	}
-	urlObj.Path = path.Join(urlObj.Path, "index")
+	if devfileType == indexSchema.StackDevfileType {
+		urlObj.Path = path.Join(urlObj.Path, "index")
+	} else if devfileType == indexSchema.SampleDevfileType {
+		urlObj.Path = path.Join(urlObj.Path, "index", "sample")
+	} else {
```

Comment:
Will this line work when the only types for `indexSchema.DevfileType` are stack or sample? Shouldn't we have one for both?

Reply:
Would like to point out that I dont think the Index schema devfile type is mandatory.. We should probably also enforce that IMO

Reply:
Yeah, agreed.

Reply:
> Will this line to get both stacks and samples work when the only types for `indexSchema.DevfileType` are stack or sample?

Yeah, it works. The current behavior is:
If user specify the devfile type -> return the list with the devfile type
If user doesn't specify the devfile type -> return the list contains all devfile types

> Would like to point out that I dont think the Index schema devfile type is mandatory.. We should probably also enforce that IMO

Currently user can pass `""` to the function, that means no devfile type is specified. But I think you mean to make the function argument `devfileType` optional?

Reply:
No my argument is, index should always have `sample`/`stack`, it should not allow `""`

Reply:
Ok, if that's the case, we can one more devfile type called `all`, how do you think?

Reply:
Discussed with Maysun over the slack, we decide to use variadic functions like `func GetRegistry(registry string, devfileType ...indexSchema.DevfileType) ([]indexSchema.Schema, error)` instead of adding new devfile type `all` because devfile type will be used and displayed in index.json and other metadata file, devfile type `all` doesn't make sense in those files.

Reply:
Updated